### ISSUE TITLE
"mutex lock failed" error

### ISSUE THREAD

**Issue by semorrison** [leanprover/lake#21]
I reported this in https://github.com/leanprover/lake/issues/17, but then it was mixed in with another problem.

I'm now seeing

```
libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument
```

with lake master again. The crash report is [here](https://gist.github.com/semorrison/02dbcbf3d72eb995246cc0082df6b5a9).

**Comment by tydeu** [comment]
@semorrison  Is this a regression? That is, does this only appear in Lake master, and not with the earlier previous version we were testing `mathport` with (i.e., 86a55d3881b0b11d285041abe3ef2f618d696190)?

**Comment by tydeu** [comment]
If this appears in both versions then this part might be a Lean bug not a Lake bug. 

From the crash report,  It does appear to be occurring inside `IO.wait` itself with some mutex that is attempting to lock. I would think it would have a prettier Lean style error if this was a user-level bug. @Kha what do you think?

**Comment by tydeu** [comment]
Even if this is in fact a Lake bug, it is going to be much harder to debug than #17 without some more informative error on the Lean side, because there is no hint as to the state of the program leading up to this crash (whereas #17 had the repeated update Git dependency message).

**Comment by semorrison** [comment]
Unfortunately it's also apparently a Heisenbug. I haven't even been able to reproduce it.

**Comment by tydeu** [comment]
@semorrison 

> No, this is not a regression --- this error had turned up once amongst the other errors, as reported in #17.

Ah, so this only the second time you have ever seen this? 

Since part of #17 was, in fact, a Lake bug, what I was asking was whether this bug had disappeared after the fix for that bug (i.e., d81d0bae45c51b3e875500dcd595bb5d8676482b) and then reappeared only when upgrading to the current master. However, if this bug appears very infrequently then that would be unfortunately very difficult to test.

Thus, as it currently stands, it sounds this bug will be impossible to fix unless one of has some flash of insight into its cause or some reliable way to reproduce it is discovered.

**Comment by tydeu** [comment]
Wow! It looks like the Lean devs already figured out what was going on here and fixed it with 106adb09b920fdb2248750da898e07e633d576a0 (at least based on that commit's message)!

**Comment by leodemoura** [comment]
@tydeu Yes, the commit above should fix the "mutex lock failed" error.
